src/chatd.cpp

```diff
@@ -519,14 +525,13 @@ void Connection::reset() //immediate disconnect
 
 bool Connection::sendBuf(Buffer&& buf)
 {
-    if (!isOnline())
+    if (!isLoggedIn() && !isConnected())
         return false;
 //WARNING: ws_send_msg_ex() is destructive to the buffer - it applies the websocket mask directly
 //Copy the data to preserve the original
     auto rc = ws_send_msg_ex(mWebSocket, buf.buf(), buf.dataSize(), 1);
     buf.free(); //just in case, as it's content is xor-ed with the websock datamask so it's unusable
-    bool result = (!rc && isOnline());
-    return result;
+    return rc;
```

Comment:
Taking into account the fact that `libws` runs in one thread and the library runs in a different thread, relying on the connection state to return success or failure doesn't seem correct to me. If `libws` says the the message was sent successfully, why should we return a failure? Just because we have changed the status of the connection? 
Remember that a message already sent to the TPC/IP stack will still be sent if there's no connectivity, when it's back.
So here the point is to analyze what's better: return failure when the function succeed, but the library thinks it's offline, or rely on the result of the function blindly. What would be the consequence of each option?? @alxvasilev